### PR TITLE
Import alignment

### PR DESCRIPTION
**Pull request by jmbarzee**
## What changed?
Make import formatting a part of the build, and test validity with "ensure-no-changes" target

## Why?
So we never discuss import order again (hopefully)

## How did you test it?
its a build change, so the build will test it!

## Potential risks
As long as build works, the only thing I risk is the ire of coworker who want imports to be different ;)

## Documentation
N/A

## Is hotfix candidate?
No

### CODE REVIEW

**Review by carlydf — CHANGES_REQUESTED** [review]
please answer the what/why questions in the PR description :)

File: Makefile
Comment (dnr) [inline Makefile:302]:
```suggestion
	@UNGENERATED_FILES=$$(find . -type f -name '*.go' -print0 | xargs -0 grep -L -e "Code generated by .* DO NOT EDIT." || true) && \
```

File: Makefile
Comment (dnr) [inline Makefile:301]:
```suggestion
	@printf $(COLOR) "Run goimports for all files..."
```

**Comment by dnr** [comment]
Is the intention that `go.temporal.io/api` and `go.temporal.io/server` should be together in the same section or not? Because many(most?) of the changed files have them in separate sections here

File: Makefile
Comment (jmbarzee) [reply Makefile:302]: Excellent, forgot this was a grep option.

**Comment by carlydf** [comment]
> Is the intention that `go.temporal.io/api` and `go.temporal.io/server` should be together in the same section or not? Because many(most?) of the changed files have them in separate sections here

Anecdotally I've been putting them in separate sections my whole time here; I forget who told me to do that but I know someone did. So I think separating them is intentional

**Review by carlydf — APPROVED** [review]

**Review by carlydf — APPROVED** [review]
approved so you can pounce on the merge button as soon as it's ready, but I'd still like to see David's "Run goimports for all files..." suggestion included

**Comment by dnr** [comment]
> > Is the intention that `go.temporal.io/api` and `go.temporal.io/server` should be together in the same section or not? Because many(most?) of the changed files have them in separate sections here
> 
> Anecdotally I've been putting them in separate sections my whole time here; I forget who told me to do that but I know someone did. So I think separating them is intentional

We should make a team-level call on that one before merging. We have been doing it separate, Jacob said he wants to do it together, but let's just discuss and pick one (and fix this PR to do it correctly)

**Comment by jmbarzee** [comment]
I would prefer separate, I think that was the configuration that I saw when talking to most people and I think that's the preference I have. 

The catch is that we can't actually choose together or separate. AFAICT, goimports (and the tooling in goland) only organizes imports that do not have an empty line between them. Soooo, you actually can get around this by adding new lines to enforce your own preferred order. 

We can see this as a feature or a bug. In my opinion, its both, because it lets us separate `api` and `server` manually, if we want to. 

In terms of the tooling, the only thing we get to configure here is: "what special prefix(es) would you like to separate from other imports. I chose `go.temporal.io` because then we separate all temporal files. if we instead choose `go.temporal.io` then the api files will be grouped with all other 3rd party imports.

 I will go in-depth explaining how the sorting works if you would like, but you are better off just trying to run the new make command yourself.

**Review by carlydf — APPROVED** [review]
looks good to me

**Pull request merged**
